diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4eb497c..c6424fc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -43,17 +43,17 @@ IF ( WIN32 )
 ELSE  ( WIN32 )
   ## Unix and OS X...
   IF ( APPLE )
-    SET ( CMAKE_OSX_ARCHITECTURES "x86_64;i386" )
+    SET ( CMAKE_OSX_ARCHITECTURES "x86_64")
     FIND_LIBRARY ( COCOA_LIBRARY Cocoa )
     INCLUDE_DIRECTORIES ( macosx )
-    INCLUDE_DIRECTORIES ( /Library/Frameworks/Tk.framework/Versions/8.6/PrivateHeaders )
-    INCLUDE_DIRECTORIES ( /System/Library/Frameworks/Tk.framework/Versions/8.5/Headers/tk-private )
-    INCLUDE_DIRECTORIES ( /System/Library/Frameworks/Tk.framework/Versions/8.5/Headers/tk-private )
+#   INCLUDE_DIRECTORIES ( /Library/Frameworks/Tk.framework/Versions/8.6/PrivateHeaders )
+#   INCLUDE_DIRECTORIES ( /System/Library/Frameworks/Tk.framework/Versions/8.5/Headers/tk-private )
+#   INCLUDE_DIRECTORIES ( /System/Library/Frameworks/Tk.framework/Versions/8.5/Headers/tk-private )
     ADD_DEFINITIONS     ( -DMAC_TK_COCOA -DMAC_OSX_TK)
     ADD_DEFINITIONS     ( -DMAC_OSX_TK )
     ADD_DEFINITIONS     ( -std=gnu99 )
     ADD_DEFINITIONS     ( -x objective-c )
-    ADD_DEFINITIONS     ( -fobjc-gc  )
+#   ADD_DEFINITIONS     ( -fobjc-gc  )
     ADD_DEFINITIONS     ( -fno-objc-arc )
 #   ADD_DEFINITIONS     ( -fobjc-arc )
     LINK_LIBRARIES      ( ${COCOA_LIBRARY} )
@@ -125,7 +125,7 @@ SET ( CP           ${CMAKE_COMMAND} -E copy )
 ##  Locate Tcl/Tk
 ## ===========================================================================
 MESSAGE ( STATUS "Searching for Tcl/Tk..." )
-FIND_PACKAGE ( TCL REQUIRED )
+#FIND_PACKAGE ( TCL REQUIRED )
 FIND_PACKAGE ( TclStub REQUIRED )
 
 ## Tcl/Tk info (useful for debug purposes)...
